airbyte-integrations/connectors/source-salesforce/source_salesforce/api.py

```diff
@@ -308,9 +310,12 @@ def login(self):
             "client_secret": self.client_secret,
             "refresh_token": self.refresh_token,
         }
-
-        resp = self._make_request("POST", login_url, body=login_body, headers={"Content-Type": "application/x-www-form-urlencoded"})
-
+        try:
+            resp = self._make_request("POST", login_url, body=login_body, headers={"Content-Type": "application/x-www-form-urlencoded"})
+        except HTTPError as err:
+            if err.response.status_code == requests.codes.BAD_REQUEST:
```

Comment:
can we also check the error message and only ask the user to re-authenticate if it says their token is expired?

It would be possible for us to introduce a bug that results in 400 errors, and we wouldn't want to raise it as a configuration error

Reply:
updated